addons/website_event/static/tests/tours/website_event.js

```diff
@@ -1,60 +1,97 @@
 /** @odoo-module **/
 
-    import { _t } from "@web/core/l10n/translation";
-    import wTourUtils from "@website/js/tours/tour_utils";
+import wTourUtils from "@website/js/tours/tour_utils";
 
-    import { markup } from "@odoo/owl";
+function websiteCreateEventTourSteps() {
+    return [
+        {
+            content: "Click here to add new content to your website.",
+            trigger: ".o_menu_systray .o_new_content_container > a",
+            consumeVisibleOnly: true,
+        },
+        {
+            content: "Click here to create a new event.",
+            trigger: "a[data-module-xml-id='base.module_website_event']",
+        },
+        {
+            content:
+                "Create a name for your new event and click `Continue`. e.g: Technical Training",
+            trigger: ".modal-dialog div[name='name'] input",
+            run: "text Technical Training",
+        },
+        {
+            content: "Open date range picker. Pick a Start date for your event",
+            trigger: ".modal-dialog div[name=date_begin]",
+            run: () => {
+                document.querySelector("input[data-field='date_begin']").value =
+                    "09/30/2020 08:00:00";
+                document
+                    .querySelector("input[data-field='date_begin']")
+                    .dispatchEvent(new Event("change"));
 
-    wTourUtils.registerWebsitePreviewTour("website_event_tour", {
+                document.querySelector("input[data-field='date_end']").value =
+                    "10/02/2020 23:00:00";
+                document
+                    .querySelector("input[data-field='date_end']")
+                    .dispatchEvent(new Event("change"));
+
+                document.querySelector("input[data-field='date_begin']").click();
+            },
+        },
+        {
+            content: "Click `Continue` to create the event.",
+            trigger: ".modal-footer button.btn-primary",
+            extra_trigger: ".modal-dialog input[type=text][value!='']",
+        },
+        {
+            content: "Drag this block and drop it in your page.",
+            trigger:
+                "#oe_snippets.o_loaded #snippet_structure .oe_snippet:eq(2) .oe_snippet_thumbnail",
+            run: "drag_and_drop_native iframe #wrapwrap > main",
+        },
+        {
+            content: "Once you click on save, your event is updated.",
+            trigger: "button[data-action=save]",
+            // Wait until the drag and drop is resolved (causing a history step)
+            // before clicking save.
+            extra_trigger:
+                ".o_we_external_history_buttons button[data-action=undo]:not([disabled])",
+        },
+        {
+            content: "Click to publish your event.",
+            trigger: ".o_menu_systray_item .o_switch_danger_success",
+            extra_trigger: "iframe body:not(.editor_enable)",
+        },
+    ];
+}
+
+function websiteEditEventTourSteps() {
```

Comment:
What is thew need of creating functions here? can't we simply append new steps at the bottom of the tour? is it intentionally added to reuse the steps?

Reply:
Perhaps @shsa-odoo is the best person to clarify this. You can also refer to this discussion for more details: https://github.com/odoo/odoo/pull/179720#discussion_r1760590563.

Reply:
Hello @msh-odoo initially we began with writing testcase in different files, then i suggested @panv-odoo to centralize all the testcase related to website_event, and separate the testing steps and for better clearity while keeping them in the same place. Please refer https://github.com/odoo/odoo/pull/179720#discussion_r1762330486. Thanks!

Reply:
Also @panv-odoo , MSH could be asking about why we need to create function, and not simply set of steps?